promql/engine.go

```diff
@@ -1449,6 +1452,79 @@ func (ev *evaluator) rangeEvalAgg(ctx context.Context, aggExpr *parser.Aggregate
 	return result, warnings
 }
 
+// expandSeriesToMatrix generates a Matrix between ev.startTimestamp and ev.endTimestamp (inclusive), each point spaced ev.interval apart, from series given offset.
+// For every storage.Series iterator in series, the method iterates in ev.interval sized steps from ev.startTimestamp until and including ev.endTimestamp,
+// collecting every corresponding sample (obtained via ev.vectorSelectorSingle) into a Series.
+// All of the generated Series are collected into a Matrix, that gets returned.
+func (ev *evaluator) expandSeriesToMatrix(ctx context.Context, series []storage.Series, offset time.Duration, recordOrigT bool) Matrix {
```

Comment:
In #14900 this was renamed and redesigned; don't we want that change here?

Reply:
I think the issue is that in this PR, the method changes to taking a `[]storage.Series` argument instead of a `parser.VectorSelector` argument. This was necessary for the method's use within the `fetchInfoSeries` method. As such, the `evalVectorSelector` name doesn't make any sense any longer (there's no `VectorSelector` to eval).

Reply:
On second thought, maybe the method could be called `evalSeries`? Considering it would previously operate on the `VectorSelector`'s series. Do you agree?

Reply:
So you don't want #14900 any more?

Reply:
> So you don't want https://github.com/prometheus/prometheus/pull/14900 any more?

#14900 is in no way invalidated by this PR. The method name and the signature change, but the basics remain. Would you prefer if I make another PR to release-2.55, introducing the same refactoring as in #14900? That would make it easier to reason about this PR I think.

Reply:
> Backport: https://github.com/prometheus/prometheus/pull/15163

Oh, didn't see this at first. 👀

Reply:
Rebased - I also renamed the method to `evalSeries` as I suggested above, since I think it makes sense (the argument type changing to `[]storage.Series`).